app/views/shared/_cash_transactions.html.erb

```diff
@@ -0,0 +1,32 @@
+<% @cash_transaction_types.each do |type| %>
+
+  <div class="govuk-!-padding-bottom-4"></div>
+
+  <h2 class="govuk-heading-m"><%= t('.cash_heading') %> <%= t(".#{type}") %></h2>
+
+  <%= govuk_hint t('.cash_hint') %>
+
+  <table class="govuk-table" style="margin-bottom:0px;">
+    <thead class="govuk-table__head">
+    <tr class="govuk-table__row">
+      <th class="govuk-table__header" scope="col"><%= t('.month') %></th>
+      <th class="govuk-table__header govuk-table__cell--numeric govuk-!-font-weight-bold" scope="col"><%= t('.amount') %></th>
+    </tr>
+    </thead>
+
+    <tbody class="govuk-table__body">
+    <% @cash_transactions.select {|i| i.transaction_type.name == type}.each do |transaction| %>
+      <tr class="govuk-table__row">
+        <td class="govuk-table__cell"><%= t('.dates', date_end: l(transaction.transaction_date.at_end_of_month , format: :short_date), date_start: l(transaction.transaction_date.at_beginning_of_month, format: :short_date)) %></td>
+        <td class="govuk-table__cell govuk-table__cell--numeric"><%= transaction.amount %></td>
+      </tr>
+    <% end %>
+    <tr class="govuk-table__row">
+      <td class="govuk-table__cell"><strong><%= t('.total') %></strong></td>
+      <td class="govuk-table__cell govuk-table__cell--numeric"><strong>
+        <%= number_to_currency CashTransaction.amounts_for_application(@legal_aid_application.id).fetch(type, 0) %>
```

Comment:
Should CashTransaction.amounts_for_application **be tested** **for when it doesn't exist** and then when the value is zero e.g. provider did not declare maintenance_in so it should show maintenance_in 'None declared' on the page and not £0

however, if the provider says maintenance_in: £0 then it should show maintenance_in £0 on the page

similar to this jira ticket AP-1934

my thinking is that, when CashTransaction.amounts_for_application(@legal_aid_application.id)**.fetch(type)** returns {} then it replaces it with the value 0 because here it says .fetch(type, **0**)

I think elsewhere in our app when the value wasn't given we replace it with the text 'None declared'  ... might want to double check that. When it is given and it is set to £0 then we say '£0' etc.

Reply:
I realised we have validation on the cash income / cash outgoings pages so that each cash_transaction must have values ^ so ignore the above comment, sorry.

**Q**: Do you need to have .fetch(type, 0) could it just be .fetch(type) ?